src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,184 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
+    public MyLinkedList() {
+        first = null;
+        last = null;
+    }
+
+    private Node<T> nodeByIndex(int index) {
+        Node<T> newNode;
+        if (index < size / 2) {
+            newNode = first;
+            for (int i = 0; i < index; i++) {
+                newNode = newNode.next;
+            }
+            return newNode;
+        } else {
+            newNode = last;
+            for (int i = size - 1; i > index; i--) {
+                newNode = newNode.prev;
+            }
+            return newNode;
+        }
+    }
+
+    private void addFirst(T value) {
+        Node<T> newNode = new Node(value);
+        first = last = newNode;
+    }
+
+    private void addFront(T value) {
+        Node<T> newNode = new Node(value);
+        first.prev = newNode;
+        newNode.next = first;
+        newNode.prev = null;
+        first = newNode;
+    }
+
+    private void addBack(T value) {
+        Node<T> newNode = new Node(value);
+        newNode.prev = last;
+        last.next = newNode;
+        newNode.next = null;
+        last = newNode;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == 0) {
+            addFirst(value);
+        } else {
+            addBack(value);
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size < index || index < 0) {
+            throw new IndexOutOfBoundsException();
+        }
+        if (index == 0) {
+            if (size == 0) {
+                addFirst(value);
+            } else {
+                addFront(value);
+            }
+        }
+        if (index == size) {
+            addBack(value);
+        }
+        Node<T> newNode = new Node(value);
+        Node<T> node = nodeByIndex(index - 1);
+        node.next = newNode;
```

Comment:
does is work correctly?
newNode.next = node.prev;
        newNode.prev = node;

Reply:
fixed